internal/plugins/ansible/v1/api.go

```diff
@@ -39,7 +40,8 @@ const (
 )
 
 type createAPIPSubcommand struct {
-	config        *config.Config
+	config        config.Config
+	options       *golang.Options
```

Comment:
We discussed a couple days ago that probably most of the validation that is done in `Options` should be moved into `Resource` in kb. I made a PR for this in https://github.com/kubernetes-sigs/kubebuilder/pull/1974, probably merging that and using `resource.Validate()` should be the way to go.

Reply:
In the same topic https://github.com/kubernetes-sigs/kubebuilder/pull/1986 will also offer a better API for constructing new `Resource` objects instead of having to use the `Options` object that represents command flag options for specific plugins.

Reply:
I added a commented where it is used. It will be a follow-up.

Reply:
Since `golang.Options.Validate()` is a net-new feature, I'd rather see the change to `resource.Validate()` made in this PR.